### PR TITLE
ensure temporary files are placed in the temporary directory

### PR DESCRIPTION
**Pull request by dhellmann**
Add --tmpdir argument to ensure they go to ${TMPDIR:-/tmp}.

### CODE REVIEW

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1680/

**Comment by hardys** [comment]
Sorry @dhellmann I merged https://github.com/openshift-metal3/dev-scripts/pull/936 so this needs a rebase, I figured if we're moving the location we need to be robust with the cleanup, so we'll need to ensure the trap removes the files from the tmpdir here.

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1687/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1693/

**Comment by metal3ci** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/1706/

File: 04_setup_ironic.sh [hunk 1]
```diff
@@ -17,17 +17,23 @@ source utils.sh
...
 jq -s '.[0] * .[1]' ${REGISTRY_AUTH_FILE} ${REGISTRY_CREDS} > ${COMBINED_AUTH_FILE}
 
 DOCKERFILE=$(mktemp "release-update--XXXXXXXXXX")
 _tmpfiles="$_tmpfiles $DOCKERFILE"
+{ echo "${PULL_SECRET}" ; } 2> /dev/null > $REGISTRY_AUTH_FILE
+
+# Combine pull-secret with registry's password
+jq -s '.[0] * .[1]' ${REGISTRY_AUTH_FILE} ${REGISTRY_CREDS} > ${COMBINED_AUTH_FILE}
+
+DOCKERFILE=$(mktemp --tmpdir "release-update--XXXXXXXXXX")
```
Comment (derekhiggins) [inline 04_setup_ironic.sh:36]: Looks like a rebase problem here, we are setting "DOCKERFILE" twice and a couple of other lines.

Comment (dhellmann) [reply 04_setup_ironic.sh]: Fixed

File: 04_setup_ironic.sh [hunk 2]
```diff
@@ -17,17 +17,19 @@ source utils.sh
...
-COMBINED_AUTH_FILE=$(mktemp "combined-pullsecret--XXXXXXXXXX")
+COMBINED_AUTH_FILE=$(mktemp --tmpdir "combined-pullsecret--XXXXXXXXXX")
 _tmpfiles="$_tmpfiles $COMBINED_AUTH_FILE"
 jq -s '.[0] * .[1]' ${REGISTRY_AUTH_FILE} ${REGISTRY_CREDS} > ${COMBINED_AUTH_FILE}
 
-DOCKERFILE=$(mktemp "release-update--XXXXXXXXXX")
+DOCKERFILE=$(mktemp --tmpdir "release-update--XXXXXXXXXX")
 _tmpfiles="$_tmpfiles $DOCKERFILE"
+{ echo "${PULL_SECRET}" ; } 2> /dev/null > $REGISTRY_AUTH_FILE
+
```
Comment (derekhiggins) [inline 04_setup_ironic.sh:32]: Extra lines here (from line 22)

Comment (dhellmann) [reply 04_setup_ironic.sh]: This series is a mess! Fixed.

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1708/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1714/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1723/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1727/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1748/

**Pull request merged**